esphome/core/component.cpp

```diff
@@ -231,10 +241,11 @@ void PollingComponent::set_update_interval(uint32_t update_interval) { this->upd
 WarnIfComponentBlockingGuard::WarnIfComponentBlockingGuard(Component *component)
     : started_(millis()), component_(component) {}
 WarnIfComponentBlockingGuard::~WarnIfComponentBlockingGuard() {
-  uint32_t now = millis();
-  if (now - started_ > 50) {
+  uint32_t blocking_time = millis() - started_;
+  if ((component_ != nullptr) ? component_->should_warn_of_blocking(blocking_time)
+                              : (blocking_time > WARN_IF_BLOCKING_OVER_MS)) {
```

Comment:
```suggestion
  bool some_condition = blocking_time > WARN_IF_BLOCKING_OVER_MS;
  if (this->component_ != nullptr) {
    some_condition = this->component_->should_warn_of_blocking(blocking_time);
  } 
  if (some_condition) {
```

```suggestion
  bool some_condition;
  if (this->component_ != nullptr) {
    some_condition = this->component_->should_warn_of_blocking(blocking_time);
  } else {
    some_condition = blocking_time > WARN_IF_BLOCKING_OVER_MS;
  } 
  if (some_condition) {
```
Give it proper name. `some_condition` is just an example.